pkgs/applications/editors/vim/plugins/nvim-treesitter/overrides.nix

```diff
@@ -15,30 +14,47 @@ self: super:
 let
   inherit (neovimUtils) grammarToPlugin;
 
-  overrides = prev: {
+  buildQueries =
+    { language }:
+    runCommand "nvim-treesitter-queries-${language}"
+      {
+        passthru = { inherit language; };
+        meta.description = "Queries for ${language} from nvim-treesitter";
+      }
+      ''
+        mkdir -p "$out/queries"
+        if [ -d "${super.nvim-treesitter.src}/runtime/queries/${language}" ]; then
+          ln -s "${super.nvim-treesitter.src}/runtime/queries/${language}" "$out/queries/${language}"
+        else
+          mkdir -p "$out/queries/${language}"
+        fi
+      '';
+
+  generated = callPackage ./generated.nix {
+    inherit (tree-sitter) buildGrammar;
+    inherit buildQueries;
   };
 
-  generatedGrammars =
-    let
-      generated = callPackage ./generated.nix {
-        inherit (tree-sitter) buildGrammar;
-      };
-    in
-    lib.overrideExisting generated (overrides generated);
+  inherit (generated) parsers queries;
 
-  generatedDerivations = lib.filterAttrs (_: lib.isDerivation) generatedGrammars;
+  parsersWithMeta = builtins.mapAttrs (
+    lang: parser:
+    if builtins.hasAttr lang queries then
+      parser.overrideAttrs (old: {
+        passthru = (old.passthru or { }) // {
+          associatedQuery = queries.${lang};
+        };
+      })
+    else
+      parser
+  ) parsers;
 
-  # add aliases so grammars from `tree-sitter` are overwritten in `withPlugins`
-  # for example, for ocaml_interface, the following aliases will be added
-  #   ocaml-interface
-  #   tree-sitter-ocaml-interface
-  #   tree-sitter-ocaml_interface
```

Comment:
question: Why has this comment been deleted? It looks like we're still doing this.